test/pkg/environment/common/expectations.go

```diff
@@ -80,9 +79,6 @@ func (env *Environment) ExpectUpdated(objects ...client.Object) {
 		Eventually(func(g Gomega) {
 			current := o.DeepCopyObject().(client.Object)
 			g.Expect(env.Client.Get(env.Context, client.ObjectKeyFromObject(current), current)).To(Succeed())
-			if current.GetResourceVersion() != o.GetResourceVersion() {
```

Comment:
Why are we choosing to drop this here? I think that it's fine to keep this function as-is, but to just call Update directly for cases when we want to directly handle the Conflict error that's coming back. Anyways, IIRC, we just need to drop the equality check and the blind patch should be enough to fix the issue here

Reply:
Sounds good, I thought we would want to make a build call even if we have a different version then expected.